src/mesh/position.cxx

```diff
@@ -0,0 +1,97 @@
+#include "bout/sys/position.hxx"
+#include "boundary_region.hxx"
+#include "bout/constants.hxx"
+#include "bout/mesh.hxx"
+
+Position::Position(const BoundaryRegion* bndry, int iz, CELL_LOC loc, BoutReal _t, Mesh * msh)
+  : iz(iz), _t(_t), msh(msh), sx(loc == CELL_XLOW), sy(loc == CELL_YLOW), sz(loc == CELL_ZLOW) {
+  ix = bndry->x;
+  iy = bndry->y;
+  if (bndry->bx) {
+    sx = true;
+    if (bndry->bx < 0) {
+      ix += 1;
+    }
+  }
+  if (bndry->by) {
+    // iy is already outside
+    // Direction of the boundary [x+bx][y+by] is going outwards
+    // by < 0 -> same direction as staggering
+    // staggering info in boundary condition needs to be ignored
+    sy = true;
+    if (bndry->by < 0) {
+      iy += 1;
+    }
+  }
+};
+
+BoutReal Position::x() {
+  BoutReal fac = 1;
+  switch (fx) {
+  case TWOPI:
+    fac = 2 * PI;
+    BOUT_FALL_THROUGH;
+  case DEFAULT:
+  case ONE:
+    if (sx) {
+      return fac * 0.5 * (msh->GlobalX(ix) + msh->GlobalX(ix - 1));
+    } else {
+      return fac * msh->GlobalX(ix);
+    }
+  case VALUE:
+    return _x;
+  case INVALID:
+    printf("WARNING: Accessing zero value\n");
```

Comment:
`printf` -> `output_error`

Should this instead throw an exception? What does `INVALID` really imply?